include/dlaf/eigensolver/reduction_to_band.h

```diff
@@ -72,14 +28,16 @@ auto groupTausFromBandsToTiles(
 /// band-diagonal result together with the elementary reflectors. Just the tiles of the lower
 /// triangular part will be used.
 /// @param band_size size of the band of the resulting matrix (main diagonal + band_size sub-diagonals)
+/// @return a column vector of size max(0, mat_a.size().rows() - band_size - 1)
+/// (block size mat_a.blockSize.rows()) containing taus
 ///
 /// @pre mat_a has a square size
 /// @pre mat_a has a square block size
 /// @pre mat_a is a local matrix
 /// @pre mat_a.blockSize().rows() % band_size == 0
 template <Backend B, Device D, class T>
-common::internal::vector<pika::shared_future<common::internal::vector<T>>> reductionToBand(
-    Matrix<T, D>& mat_a, const SizeType band_size) {
+// TODO: Pass in reference to taus matrix?
```

Comment:
Not sure what is the advantage.

On the other hand a couple of disadvantage:
- the (advanced) user has to allocate tau with the correct distribution and size
- we have to check that tau has correct shape and distribution
- the documentation can be simple (especially with the current distribution workarounds)

Reply:
I don't have a use case for it. I was wondering if there is a use case for it since it's similar to the eigensolvers having overloads taking the eigenvalues/vectors and didn't want to forget to ask about it. I'm happy not to add it and I've removed the TODO in https://github.com/eth-cscs/DLA-Future/pull/908/commits/676ce6d0528d83e4a2df45010e62a0e1a9900df0.